client/client.go

```diff
@@ -265,15 +296,24 @@ func (c *Client) DeactivateUser(userId string, status string) error {
 		log.Println("[DEACTIVATE/ACTIVATE ERROR]: ",err)
 		return nil
 	}
-	var newdata map[string]interface{}
-	newbody, err := ioutil.ReadAll(resp.Body)
-	err = json.Unmarshal([]byte(newbody), &data)
 	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
 		return nil
-    	} else {
+    } else {
+		var newdata map[string]interface{}
+		newbody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Println("[ERROR]: ",err)
+			return err
+		}
+		err = json.Unmarshal([]byte(newbody), &data)
+		if err != nil {
+			log.Println("[ERROR]: ",err)
+			return err
+		}
 		log.Println("[DEACTIVATE/ACTIVATE ERROR]")
 		return fmt.Errorf("Error : %v",newdata["message"])
-    	}
+    }
+	
 }
 
 func (c *Client) IsRetry(err error) bool {
```

Comment:
Here `too many requests error` is handled using status code. But the error returned in the create, read, update, delete, deactivate functions contains the error message only. Hence, this function will not be able to identify and handle too many request error. 
Either add status code in the error returned by above functions or handle this error using error message.